addons/binding/org.openhab.binding.s7/src/main/java/org/openhab/binding/s7/handler/S7ThingHandler.java

```diff
@@ -0,0 +1,180 @@
+    private long updateTimeout = 5000;
+
+    public S7ThingHandler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        if (channelUID.getId().equals(CHANNEL_STATE)) {
+            if (command instanceof OnOffType) {
+                final boolean setOn = ((OnOffType) command) == OnOffType.ON;
+
+                if (setOn != lastState) {
+                    lastUpdate = System.currentTimeMillis();
+                    logger.debug("Prepare setting S7 lamp {} to {}.", getThing().getLabel(), setOn);
+
+                    Thread t = new Thread() {
+                        @Override
+                        public void run() {
+                            logger.debug("Setting S7 lamp {} to {}.", getThing().getLabel(), setOn);
+
+                            try {
+                                S7BridgeHandler bridge = (S7BridgeHandler) getBridge().getHandler();
+                                int Area = ((java.math.BigDecimal) getConfig().get(INPUT_DBAREA)).intValue();
+                                int Address = ((java.math.BigDecimal) getConfig().get(INPUT_ADDRESS)).intValue();
+
+                                switch ((String) getConfig().get(ACCESS_MODE)) {
+                                    case MODE_TOGGLE:
+                                        bridge.setBit(Area, Address, false);
+                                        Thread.sleep(TAP_OFF_DELAY);
+                                        bridge.setBit(Area, Address, true);
+                                        Thread.sleep(TAP_ON_DELAY);
+                                        bridge.setBit(Area, Address, false);
+                                        break;
+                                    case MODE_READ_WRITE:
+                                        bridge.setBit(Area, Address, setOn);
+                                        break;
+                                    case MODE_PUSHBUTTON:
+                                        bridge.setBit(Area, Address, true);
+                                        updateState(CHANNEL_STATE, OnOffType.ON);
+                                        Thread.sleep(TAP_ON_DELAY);
+                                        bridge.setBit(Area, Address, false);
+                                        updateState(CHANNEL_STATE, OnOffType.OFF);
+                                        break;
+                                }
+                            } catch (InterruptedException e) {
+                                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                                        "Could not toggle state of " + getThing().getLabel() + ".");
+                                logger.warn("{}: {}", e.getMessage(), e.getStackTrace().toString());
+                            }
+                        }
+                    };
+
+                    super.scheduler.execute(t);
```

Comment:
Is this super really needed here, I would hope not, otherwise please do not shadow fields like this.